src/main/zapHomeFiles/hud/drawer.js

```diff
@@ -12,19 +12,73 @@ Vue.component('history', {
     template: '#history-template',
     data() {
         return {
-            messages: []
+            filter: '',
+            regexEnabled: false,
+            messages: [],
+            hiddenMessageCount: 0,
+            isRegExError: false,
+            enableRegExText: I18n.t("history_enable_regex")
         }
     },
     computed: {
         timeDescendingMessages() {
             return this.messages.slice().reverse();
+        },
+        messageCount() {
+            return this.messages.length;
+        },
+        filteredMessages() {
+            const self=this,
+                  isRegex = this.regexEnabled;
+            let re;
+
+            if (isRegex){
+                try {
+                    re = new RegExp(this.filter);
+                    this.isRegExError = false;
+                }
+                catch (ex) {
+                    this.isRegExError = true;
+                    return []; //Return empty array if invalid RegEx
+                }
+            }
+
+            return this.messages.filter( message => {
+                if (self.filter.trim().length === 0) {
+                    return true;
+                }
+                if (isRegex){
+                    return re.test(message.url);
+                }
+                else{
+                    return message.url.indexOf(self.filter)>=0; 
+                }
+            });
+        },
+        historyItemsFiltered() {
+            let historyItemsFilteredString = I18n.t("history_items_filtered")
+                    .replace('{0}', this.hiddenMessageCount)
+                    .replace('{1}', this.messageCount);
+            return historyItemsFilteredString;
         }
     },
     methods: {
         messageSelected(id) {
             navigator.serviceWorker.controller.postMessage({tabId: tabId, frameId: frameId, action: "showHttpMessageDetails", tool: "history", id:id});
         }
     },
+    watch: {
+        filter() {
+            //use the next tick capability to let the filteredMessages state update displayed messages and update count after reactive update
+            this.$nextTick(function () {
+                const visibleMessages = document.querySelectorAll("#history-messages .message-tr");
+                this.hiddenMessageCount = (!this.messages) ? 0 : this.messages.length - visibleMessages.length;
+            })
+        },
+        regexEnabled(){
```

Comment:
Changing just the regex state does not affect the message that shows the number of items filtered, it keeps showing the old state even though the messages are filtered.

Reply:
@thc202 nice catch! I refactored for the reactivity to trigger on the filtered message list rather than the filter input itself which made for a better solution of handling the various use cases.